app/components/post_list/post/body/content/permalink_preview/index.tsx

```diff
@@ -36,12 +37,19 @@ const enhance = withObservables(['embedData', 'serverUrl'], ({database, embedDat
         distinctUntilChanged(),
     ) : of$(false);
 
+    const filesInfo = of$(embedData?.post?.metadata?.files || []);
```

Comment:
Why are you calculating this here instead of in the component? This has nothing to do with observables.